### PR TITLE
[FIX] mail: increase chatter max-height.

### PR DESCRIPTION
**Pull request by phenix-factory**
We go back to the fake textarea to avoid a flickering created by the
`this._textareaRef.el.style.height = "0px";`.
This height reset create an unwanted scrollbar jump that could hide the content
of the textarea (not aside mode).

task-2390354

### CODE REVIEW

File: addons/mail/static/src/components/composer_text_input/composer_text_input.js [hunk 1]
```diff
@@ -162,8 +164,10 @@ class ComposerTextInput extends Component {
      * @private
      */
     _updateHeight() {
-        this._textareaRef.el.style.height = "0px";
-        this._textareaRef.el.style.height = (this._textareaRef.el.scrollHeight) + "px";
+        // console.log(this._textareaRef.el);
+        this._textareaDummyRef.el.style.height = "0px";
+        this._textareaRef.el.style.height = (this._textareaDummyRef.el.scrollHeight) + "px";
+        this._textareaDummyRef.el.style.height = (this._textareaDummyRef.el.scrollHeight) + "px";
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.js:170]: I think you might be able to permanently leave the dummy at 0px height, as long as content is updated and height says at 0, the fake scrollHeight will read the correct value no?

File: addons/mail/static/src/components/composer_text_input/composer_text_input.xml [hunk 2]
```diff
@@ -11,6 +11,7 @@
                     />
                 </t>
                 <textarea class="o_ComposerTextInput_textarea" t-att-class="{ 'o-composer-is-compact': props.isCompact }" style="height: 0;" t-esc="composer.textInputContent" t-att-placeholder="textareaPlaceholder" t-on-focusin="_onFocusinTextarea" t-on-focusout="_onFocusoutTextarea" t-on-keydown="_onKeydownTextarea" t-on-keyup="_onKeyupTextarea" t-on-input="_onInputTextarea" t-ref="textarea"/>
+                <textarea class="o_ComposerTextInput_textarea_dummy" t-att-class="{ 'o-composer-is-compact': props.isCompact }" style="height: 0;" t-esc="composer.textInputContent" t-ref="textarea_dummy" disabled="1" />
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.xml:14]: Is it necessary/safe to set disabled=1? Any attribute that differs from the original textarea might influence the way it is rendered.

Also not having the class `o_ComposerTextInput_textarea` is problematic because it has style scss that has to be taken into account in the compte.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 3]
```diff
@@ -27,6 +28,16 @@
...
 }
+
+.o_ComposerTextInput_textarea_dummy {
+    position: absolute;
+    opacity: 0;
+    border-top-width: 0;
+    border-bottom-width: 0;
+    overflow: hidden;
+    top: -10000;
+    direction: -10000;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:42]: You probably copied it, but that doesn't look like a valid value for direction, I have no idea why we would need that

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 4]
```diff
@@ -27,6 +28,16 @@
...
-        max-height: 150px;
+        max-height: 400px;
     }
 }
+
+.o_ComposerTextInput_textarea_dummy {
+    position: absolute;
+    opacity: 0;
+    border-top-width: 0;
+    border-bottom-width: 0;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:39]: Changing the border size does not look like a good idea when computing height. Would be best if it was removed from here, and the compute method took into account `clientHeight` or `offsetHeight`, as necessary.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.js (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.js]: you are right, it work's the same.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.xml (hunk 2 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.xml]: The disable is to avoid selecting the textarea with `tab`. We don't really want this textarea edited at all by anything else than javascript.
I have add the `o_ComposerTextInput_textarea` class

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 4 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss]: Copied from `web.dom`. But work's the same if removed.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.js [hunk 5]
```diff
@@ -162,8 +164,8 @@ class ComposerTextInput extends Component {
      * @private
      */
     _updateHeight() {
-        this._textareaRef.el.style.height = "0px";
-        this._textareaRef.el.style.height = (this._textareaRef.el.scrollHeight) + "px";
+        this._textareaDummyRef.el.style.height = "0px";
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.js:167]: Do we even need this line then? If height is already set at 0 in the XML that's all we need.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss [hunk 6]
```diff
@@ -8,6 +8,7 @@
 }
 
 .o_ComposerTextInput_textarea {
+    min-height: 39px;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.scss:11]: Why is this necessary now, and why 39px?

File: addons/mail/static/src/components/composer_text_input/composer_text_input.xml [hunk 7]
```diff
@@ -11,6 +11,7 @@
                     />
                 </t>
                 <textarea class="o_ComposerTextInput_textarea" t-att-class="{ 'o-composer-is-compact': props.isCompact }" style="height: 0;" t-esc="composer.textInputContent" t-att-placeholder="textareaPlaceholder" t-on-focusin="_onFocusinTextarea" t-on-focusout="_onFocusoutTextarea" t-on-keydown="_onKeydownTextarea" t-on-keyup="_onKeyupTextarea" t-on-input="_onInputTextarea" t-ref="textarea"/>
+                <textarea class="o_ComposerTextInput_textarea o_ComposerTextInput_textarea_dummy" t-att-class="{ 'o-composer-is-compact': props.isCompact }" style="height: 0;" t-esc="composer.textInputContent" t-ref="textarea_dummy" disabled="1" />
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_text_input/composer_text_input.xml:14]: We can probably remove both height: 0 from the xml now, if you add height 0 on the css of the dummy.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 4 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss]: Make sure the code is robust enough to also work if we change the border size and/or padding of the textarea to either 0 or non-0 values.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.xml (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.xml]: Ok if disable is to prevent tab that sounds important. Let's hope it doesn't change style in a way that would break height computation.

File: addons/mail/static/src/components/composer_text_input/composer_text_input.scss (hunk 6 above)
Comment (phenix-factory) [reply addons/mail/static/src/components/composer_text_input/composer_text_input.scss:11]: It fixed an issue I don't have anymore. I guess other cleanup also fix this size issue.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Looks good otherwise

**Comment by alexkuhn** [comment]
LGTM. `ci/upgrade_enterprise` failed for some reasons. Maybe rebase?

**Comment by phenix-factory** [comment]
@alexkuhn Green !

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
